uclalib_get_ssh_host_keys.yml

```diff
@@ -0,0 +1,100 @@
+---
+# Query every host in inventory.ini for ssh host keys, populate
+# /etc/ssh/known_hosts
+#
+# Requires python-netaddr and python-dns to be installed (RHEL7)
+
+- name: uclalib_get_ssh_host_keys.yml
+  hosts: all
+  user: ansible
+  connection: local
+  gather_facts: false
+
+  vars:
+    working_file: "/tmp/known-hosts"
+    output: "/etc/ssh/ssh_known_hosts"
+    salt: "{{ 1000000 | random }}"
+
+  tasks:
+    - name: Set Working Directory
+      set_fact:
+        working_directory: "/tmp/ssh-known-hosts-{{ salt }}"
+      run_once: true
+
+    - name: Create Working Directory
+      file:
+        path: "{{ working_directory }}"
+        state: directory
+      check_mode: false
+      run_once: true
+
+    # When the inventory record is an IP, this will cause noise
+    - name: Get port from inventory (default 22), look up IP
+      set_fact:
+        ansible_ssh_port: "{{ hostvars[inventory_hostname]['ansible_ssh_port'] | default('22') }}"
+        ansible_inventory_ip: "{{ lookup('dig',inventory_hostname) }}"
+        ansible_shortname: "{{ inventory_hostname|regex_replace('\\..*') }}"
```

Comment:
Hmm, could you give me an example output of what this sets? It sounds like we'll be storing multiple facts about the port, IP/hostname on our ansible control server? If so, do these get cleaned?

Reply:
```[jhriv@p-u-ansible01 ssh-host-keys]$ ansible-playbook -i inventory.ini -l \d-w-sin\* -e output=/tmp/example-ssh-host-keys /tmp/uclalib_get_ssh_host_keys.yml  -K
BECOME password:

PLAY [uclalib_get_ssh_host_keys.yml] **************************************************************************************************************

TASK [Set Working Directory] **********************************************************************************************************************
ok: [d-w-sinaimanuscripts01.library.ucla.edu]

TASK [Create Working Directory] *******************************************************************************************************************
changed: [d-w-sinaimanuscripts01.library.ucla.edu]

TASK [Get port from inventory (default 22), look up IP] *******************************************************************************************
ok: [d-w-sinaimanuscripts01.library.ucla.edu]
ok: [d-w-sinaiauth01.library.ucla.edu]

TASK [Assemble query string (hostname vs ip)] *****************************************************************************************************
ok: [d-w-sinaimanuscripts01.library.ucla.edu]
ok: [d-w-sinaiauth01.library.ucla.edu]

TASK [Ensure ssh host key known] ******************************************************************************************************************
# 164.67.40.152:22 SSH-2.0-OpenSSH_7.4
# 164.67.40.152:22 SSH-2.0-OpenSSH_7.4
# 164.67.40.152:22 SSH-2.0-OpenSSH_7.4
# 52.11.190.248:22 SSH-2.0-OpenSSH_7.4
# 52.11.190.248:22 SSH-2.0-OpenSSH_7.4
changed: [d-w-sinaimanuscripts01.library.ucla.edu] => (item=d-w-sinaimanuscripts01.library.ucla.edu,d-w-sinaimanuscripts01,164.67.40.152 ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQD0RDlNN9ZXIb+d4eI5G9UmKKZLT7uYiwTu1rwA9jQEECaYVHRO+1NpsOxTgc/Lf4pPYSe6kOVgZbd+WXGBoOamevTAoCZip1fgBAg+yo0yOowLbRgaDZF72RrAYLyuBnt25o5YC4hToffHiTDiuTYOuUl8ju55Mf5ZyQaOMvb85hTVAK6u2fTmEKYuQ6K2J0znDLqzoZXAM8+GjvtDpf63qBbseXd/EKnAWJO4g09r63Fqe9svWj6Dlyvvp9Tb30rnbZUPmAT0KSLKwvLktzhLIE7xWVEjj1u9GER3UI5m+W+W52o7qoC131nhyUGbdpXkDjjBtxVUdSr8+gRb/Rl1)
# 52.11.190.248:22 SSH-2.0-OpenSSH_7.4
changed: [d-w-sinaimanuscripts01.library.ucla.edu] => (item=d-w-sinaimanuscripts01.library.ucla.edu,d-w-sinaimanuscripts01,164.67.40.152 ecdsa-sha2-nistp256 AAAAE2VjZHNhLXNoYTItbmlzdHAyNTYAAAAIbmlzdHAyNTYAAABBBL8ZdjMlnX3rNPUPN1PK/H+wRSHOsOFTMpP9wgMHVdP65EITykzOS6Xm+NaCec8Env3RPwFtyUg7qQwoh8/spU0=)
changed: [d-w-sinaimanuscripts01.library.ucla.edu] => (item=d-w-sinaimanuscripts01.library.ucla.edu,d-w-sinaimanuscripts01,164.67.40.152 ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIKzIOMUQsGGm+tMg1zQmPtbYD3fpjtVwPX8J2s0+WuRl)
changed: [d-w-sinaiauth01.library.ucla.edu] => (item=d-w-sinaiauth01.library.ucla.edu,d-w-sinaiauth01,52.11.190.248 ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDG67lg3Npl70uKmziCbp4MPtw9wfqryPK2Xp2ooTw0/1Z1lzgWtt+AshC1GjcFNgzAp5hhRlZpIRntBa4AFMK86ik/YOg1JkjVKpr8oOW7iV5SdG5RZm7kG5Rss6en02OCqF/EBMP3PzSldpAEAl2qe7jgVUrDMyJPxTb3DP4qpC5UY5FyfWT1F5DRzBc36H6My0JefHTl4o1gaE90Qgl6FQVOzMARcF5/FVbAQEDKbGjpMM5TC/o+ijxe6hmxjw0db2METMJIinBJZFUpuazsDke6ThVr8tcKG1o2pDZ1Xj8ZLMN5sPpJz2sOwWoOOXmhiLN4CPZr2q5k4qVJoYqF)
changed: [d-w-sinaiauth01.library.ucla.edu] => (item=d-w-sinaiauth01.library.ucla.edu,d-w-sinaiauth01,52.11.190.248 ecdsa-sha2-nistp256 AAAAE2VjZHNhLXNoYTItbmlzdHAyNTYAAAAIbmlzdHAyNTYAAABBBNHrreu6sxYCoB7XOfYqgxojTupec3vuxKE8+R2aJJDvDUbf7Dmw0xzXHcacNsddr/OTLkHN+m8Ljou1oKn1+R0=)
changed: [d-w-sinaiauth01.library.ucla.edu] => (item=d-w-sinaiauth01.library.ucla.edu,d-w-sinaiauth01,52.11.190.248 ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAILypU6EEYyGdPyCWPHHI/SrlCCH0gmfNEk13eDJkJcRP)

RUNNING HANDLER [Assemble Host Keys] **************************************************************************************************************
changed: [d-w-sinaimanuscripts01.library.ucla.edu]

RUNNING HANDLER [Sort Host Keys] ******************************************************************************************************************
changed: [d-w-sinaimanuscripts01.library.ucla.edu]

RUNNING HANDLER [Remove Working Directory] ********************************************************************************************************
changed: [d-w-sinaimanuscripts01.library.ucla.edu]

RUNNING HANDLER [Update /tmp/example-ssh-host-keys] ***********************************************************************************************
changed: [d-w-sinaimanuscripts01.library.ucla.edu]

RUNNING HANDLER [Remove Working File] *************************************************************************************************************
changed: [d-w-sinaimanuscripts01.library.ucla.edu]

PLAY RECAP ****************************************************************************************************************************************
d-w-sinaiauth01.library.ucla.edu : ok=3    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
d-w-sinaimanuscripts01.library.ucla.edu : ok=10   changed=7    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0

[jhriv@p-u-ansible01 ssh-host-keys]$ cat /tmp/example-ssh-host-keys
d-w-sinaiauth01.library.ucla.edu,d-w-sinaiauth01,52.11.190.248 ecdsa-sha2-nistp256 AAAAE2VjZHNhLXNoYTItbmlzdHAyNTYAAAAIbmlzdHAyNTYAAABBBNHrreu6sxYCoB7XOfYqgxojTupec3vuxKE8+R2aJJDvDUbf7Dmw0xzXHcacNsddr/OTLkHN+m8Ljou1oKn1+R0=
d-w-sinaiauth01.library.ucla.edu,d-w-sinaiauth01,52.11.190.248 ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAILypU6EEYyGdPyCWPHHI/SrlCCH0gmfNEk13eDJkJcRP
d-w-sinaiauth01.library.ucla.edu,d-w-sinaiauth01,52.11.190.248 ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDG67lg3Npl70uKmziCbp4MPtw9wfqryPK2Xp2ooTw0/1Z1lzgWtt+AshC1GjcFNgzAp5hhRlZpIRntBa4AFMK86ik/YOg1JkjVKpr8oOW7iV5SdG5RZm7kG5Rss6en02OCqF/EBMP3PzSldpAEAl2qe7jgVUrDMyJPxTb3DP4qpC5UY5FyfWT1F5DRzBc36H6My0JefHTl4o1gaE90Qgl6FQVOzMARcF5/FVbAQEDKbGjpMM5TC/o+ijxe6hmxjw0db2METMJIinBJZFUpuazsDke6ThVr8tcKG1o2pDZ1Xj8ZLMN5sPpJz2sOwWoOOXmhiLN4CPZr2q5k4qVJoYqF
d-w-sinaimanuscripts01.library.ucla.edu,d-w-sinaimanuscripts01,164.67.40.152 ecdsa-sha2-nistp256 AAAAE2VjZHNhLXNoYTItbmlzdHAyNTYAAAAIbmlzdHAyNTYAAABBBL8ZdjMlnX3rNPUPN1PK/H+wRSHOsOFTMpP9wgMHVdP65EITykzOS6Xm+NaCec8Env3RPwFtyUg7qQwoh8/spU0=
d-w-sinaimanuscripts01.library.ucla.edu,d-w-sinaimanuscripts01,164.67.40.152 ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIKzIOMUQsGGm+tMg1zQmPtbYD3fpjtVwPX8J2s0+WuRl
d-w-sinaimanuscripts01.library.ucla.edu,d-w-sinaimanuscripts01,164.67.40.152 ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQD0RDlNN9ZXIb+d4eI5G9UmKKZLT7uYiwTu1rwA9jQEECaYVHRO+1NpsOxTgc/Lf4pPYSe6kOVgZbd+WXGBoOamevTAoCZip1fgBAg+yo0yOowLbRgaDZF72RrAYLyuBnt25o5YC4hToffHiTDiuTYOuUl8ju55Mf5ZyQaOMvb85hTVAK6u2fTmEKYuQ6K2J0znDLqzoZXAM8+GjvtDpf63qBbseXd/EKnAWJO4g09r63Fqe9svWj6Dlyvvp9Tb30rnbZUPmAT0KSLKwvLktzhLIE7xWVEjj1u9GER3UI5m+W+W52o7qoC131nhyUGbdpXkDjjBtxVUdSr8+gRb/Rl1
[jhriv@p-u-ansible01 ssh-host-keys]$```

Reply:
The facts set are stored only for the run, they are not saved in any files anywhere.